File: code_samples/lua/functional_map_reduce.lua
-- Functional: Map Reduce
-- AI/ML Training Sample

MapReduce = {}
MapReduce.__ index = MapReduce

function MapReduce:new()
    local obj = {
        data = ""
    }
    setmetatable(obj, self)
    return obj
end

function MapReduce:process(input)
    self.data = input
end

function MapReduce:getData()
    return self.data
end

function MapReduce:validate()
    return #self.data > 0
end

-- Example usage
local instance = MapReduce:new()
instance:process("example")
print("Data: " .. instance:getData())
print("Valid: " .. tostring(instance:validate()))

File: code_samples/dart/functional_map_reduce.dart
// Functional: Map Reduce
// AI/ML Training Sample

class MapReduce {
  String _data = '';
  
  void process(String input) {
    _data = input;
  }
  
  String getData() => _data;
  
  bool validate() => _data.isNotEmpty;
}

void main() {
  final instance = MapReduce();
  instance.process('example');
  print('Data: ${instance.getData()}');
  print('Valid: ${instance.validate()}');
}
